src/v/cluster/topic_updates_dispatcher.cc

```diff
@@ -49,9 +49,7 @@ ss::future<std::error_code> topic_updates_dispatcher::do_topic_delete(
     std::optional<assignments_set> topic_assignments;
 
     bool local_delete = transition.mode
-                          == topic_lifecycle_transition_mode::oneshot_delete
-                        || transition.mode
-                             == topic_lifecycle_transition_mode::pending_gc;
+                        != topic_lifecycle_transition_mode::drop;
```

Comment:
kinda seems like something there should be a test for if we missed it originally